internal/grpc/services/sharesstorageprovider/sharesstorageprovider.go

```diff
@@ -939,3 +942,45 @@ func (s *service) rejectReceivedShare(ctx context.Context, receivedShare *collab
 
 	return errtypes.NewErrtypeFromStatus(res.Status)
 }
+
+func (s *service) fetchShares(ctx context.Context) ([]*collaboration.ReceivedShare, map[string]string, error) {
+	lsRes, err := s.gateway.ListReceivedShares(ctx, &collaboration.ListReceivedSharesRequest{
+		// FIXME filter by received shares for resource id - listing all shares is tooo expensive!
+	})
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "sharesstorageprovider: error calling ListReceivedSharesRequest")
+	}
+	if lsRes.Status.Code != rpc.Code_CODE_OK {
+		return nil, nil, fmt.Errorf("sharesstorageprovider: error calling ListReceivedSharesRequest")
+	}
+	receivedShares := lsRes.Shares
+
+	var shareEtags map[string]string
+	if lsRes.Opaque != nil {
+		if entry, ok := lsRes.Opaque.Map["etags"]; ok {
+			// If we can't get the etags thats fine, just continue.
+			_ = json.Unmarshal(entry.Value, &shareEtags)
+		}
+	}
+	return receivedShares, shareEtags, nil
+}
+
+func findEarliestShare(receivedShares []*collaboration.ReceivedShare) (earliestShare *collaboration.Share, atLeastOneAccepted bool) {
+	atLeastOneAccepted = false
```

Comment:
This line is unnecessary, you have already defined `atLeastOneAccepted` through the named return parameter